docs/data/date-pickers/lifecycle/lifecycle.md

```diff
@@ -0,0 +1,334 @@
+---
+product: date-pickers
+title: Components lifecycle
+githubLabel: 'component: pickers'
+packageName: '@mui/x-date-pickers'
+---
+
+# Components lifecycle
+
+<p class="description">This page explains when the callbacks onChange, onAccept, and onClose are called.</p>
+
+## Fields lifecycle
+
+### On simple fields
+
+:::info
+The information below is applicable to standalone fields (when rendering `<DateField />`),
+as well as to pickers for field editing (when rendering `<DatePicker />` and using the input to edit the value).
+:::
+
+The field components have an internal state to update the visible value.
+
+It will only call the `onChange` callback when:
+
+- the user fills one section of an empty input. The value equals `Invalid date`.
+- the user completes all sections of an input. The value reflects the input.
+- the user cleans one section of a completed input. The value equals `Invalid date`.
+- the user cleans all sections of an input. The value equals `null`.
+
+In the example below, `onChange` will be called when any of the conditions from above are triggered:
+
+{{"demo": "LifeCycleDateFieldEmpty.js", "defaultCodeOpen": false}}
+
+### On range fields [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
+
+On range fields (`SingleInputDateRangeField` / `MultiInputDateRangeField` / ... ),
+`onChange` is called if the date you are modifying is matching one of the conditions above,
+even if the other date does not.
+
+In the example below, changing the value of the start date section will call `onChange` even if the end date is empty or partially filled.
+
+{{"demo": "LifeCycleDateRangeField.js", "defaultCodeOpen": false}}
+
+## Pickers lifecycle
+
+### When is `onClose` called?
+
+:::info
+If you are not controlling the `open` prop, then all the scenarios below also describe when the picker closes.
+:::
+
+#### When the last view is completed
+
+When the last view is completed, `onClose` will be called only if the `closeOnSelect` prop is equal to `true`.
```

Comment:
Strictly speaking, "completed" is more precise because it's a selection on the last view that is not a shallow selection.
But since this shallow selection does not even cause an "onChange" call, maybe we can just ignore it in the doc and use your proposal.

Reply:
Adding a suggestion as discussed.
```suggestion
When a selection in the last view is made, `onClose` will be called only if the `closeOnSelect` prop is equal to `true`.
```